examples/run-blockchain/index.js

```diff
@@ -0,0 +1,172 @@
+const async = require('async')
+const ethUtil = require('ethereumjs-util')
+const Trie = require('merkle-patricia-tree/secure')
+const Block = require('ethereumjs-block')
+const Blockchain = require('ethereumjs-blockchain')
+const BlockHeader = require('ethereumjs-block/header.js')
+const VM = require('../../')
+const Level = require('levelup')
+const Account = require('ethereumjs-account')
+const utils = require('ethereumjs-util')
+const BN = utils.BN
+const rlp = utils.rlp
+const testData = require('./test-data')
+// inMemory blockchainDB
+var blockchainDB = new Level('', { db: require('memdown') })
+
+var state = new Trie()
+
+var blockchain = new Blockchain(blockchainDB)
+blockchain.ethash.cacheDB = new Level('./.cachedb')
+
+var vm = new VM({
+  state: state,
+  blockchain: blockchain
+})
+var genesisBlock = new Block()
+
+vm.on('beforeTx', function (tx) {
+  tx._homestead = true
+})
+
+vm.on('beforeBlock', function (block) {
+  block.header.isHomestead = function () {
+    return true
+  }
+})
+
+async.series([
+  // set up pre-state
+  function (next) {
+    setupPreConditions(state, testData, next)
+  },
+
+  // create and add genesis block
+  function (next) {
+    genesisBlock.header = new BlockHeader(
+                            formatBlockHeader(
+                              testData.genesisBlockHeader
+                            )
+                          )
+    blockchain.putGenesis(genesisBlock, next)
+  },
+
+  // add some following blocks
+  function (next) {
+    async.eachSeries(testData.blocks, eachBlock, next)
+
+    function eachBlock (raw, cb) {
+      try {
+        var block = new Block(
+            new Buffer(raw.rlp.slice(2), 'hex'))
+
+        // forces the block into thinking they are homestead
+        block.header.isHomestead = function () {
+          return true
+        }
+        block.uncleHeaders.forEach(function (uncle) {
+          uncle.isHomestead = function () { return true }
+        })
+
+        blockchain.putBlock(block, function (err) {
+          cb(err)
+        })
+      } catch (err) {
+        cb()
+      }
+    }
+  },
+
+  // runBlockchain aka verify all the blocks till date
+  function runBlockchain (next) {
+    vm.runBlockchain(next)
+  },
```

Comment:
It will give a validation error too the  callback `next`. We can remove this function be shorten this to just  `vm.runBlockchain,` if wanted.